apps/web/src/app/[locale]/tracks/_components/track.action.ts

```diff
@@ -16,19 +16,22 @@ export async function enrollUserInTrack(id: number) {
   if (!session) {
     throw new Error('User is not logged in');
   }
-
-  await prisma.track.update({
-    where: {
-      id,
-    },
-    data: {
-      enrolledUsers: {
-        connect: {
-          id: session?.user.id,
+  try {
+    await prisma.track.update({
+      where: {
+        id,
+      },
+      data: {
+        enrolledUsers: {
+          connect: {
+            id: session?.user.id,
+          },
         },
       },
-    },
-  });
+    });
+  } catch {
+    throw new Error('Internal server error');
```

Comment:
why do we need a catch

Reply:
Cause there can be an error while fetching data from db , so it makes sense to add it in a try catch block and then throw the error

Reply:
if there are any other good practices I will be happy to hear it out from you : )

Reply:
youre just catching and rethrowing, wont it work the same way without it

Reply:
Wait , I understood what you are saying 
I will add a ternary to just check if user is unauthorized , else i will throw the the other error message , this will remove the try catch

Reply:
> youre just catching and rethrowing, wont it work the same way without it

yes , thats what I thought later